File: src/main/java/engine/util/utilgeometry/PointF.java
package engine.util.utilgeometry;

@SuppressWarnings("WeakerAccess")
public class PointF {
    public final float x;
    public final float y;

    public PointF() {
        this.x = 0f;
        this.y = 0f;
    }

    public PointF(final float x, final float y) {
        this.x = x;
        this.y = y;
    }

    public String toString() {
        return "(x: " + x + " ; " + "y: " + y + " )";
    }

    @Override
    public boolean equals(Object o) {

        if (o instanceof PointF) {
            return this.x == ((PointF) o).x && this.y == ((PointF) o).y;
        }

        return false;
    }
}

File: src/main/java/engine/util/utilgeometry/PointF.kt
package engine.util.utilgeometry

class PointF(val x: Float, val y: Float) {

    override fun toString(): String = "(x: $x ; y: $y )"

    override fun equals(other: Any?): Boolean = if (other is PointF) {
        this.x == other.x && this.y == other.y
    } else false
}
